File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    if (arr.length === 1) {
        return [-1];
    }

    let min = arr[0];

    for (let i = 1; i < arr.length; i++) {
        if (min > arr[i]) {
            min = arr[i];
        }
    }

    const answer = [];

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] !== min) {
            answer.push(arr[i]);
        }
    }

    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};

        if (arr.length == 1) {
            return answer = new int[]{-1};
        }
        int min = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (min > arr[i]) {
                min = arr[i];
            }
        }
        List<Integer> list = new ArrayList<>();

        for (int i = 0; i < arr.length; i++) {
            if (min != arr[i]) {
                list.add(arr[i]);
            }
        }
        answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            if (list.get(i) != min){
                answer[i] = list.get(i);
            }
        }
        return answer;
    }
}